lua/starfall/libs_sh/color.lua

```diff
@@ -67,16 +67,12 @@ end
 -- @return New color
 function instance.env.Color(r, g, b, a)
 	if isstring(r) then
-		if r[1] == "#" then r = string.sub(r, 2) end
-		if string.match(r, "%X") then
-			SF.Throw("Invalid characters in hexadecimal color", 2)
+		local hex = string.match(r, "^#?(%x+)$") or SF.Throw("Invalid hexadecimal color", 2)
+		local h2r = hex_to_rgb[#hex]
+		if h2r then
+			return wrap(h2r(tonumber(hex, 16)))
 		else
-			local h2r = hex_to_rgb[#r]
-			if h2r then
-				return wrap(h2r(r))
-			else
-				SF.Throw("Invalid hexadecimal color length", 2)
-			end
+			SF.Throw("Invalid hexadecimal color length")
```

Comment:
The 2 parameter in Throw got deleted?

Reply:
Ah sorry, must've rewritten the line and forgot about it.